### ISSUE TITLE
Vault 1.0 - PKI - Read Certificate Broken on Filesystem Storage Backend on Windows Server 2012R2

### ISSUE THREAD

**Issue by brrogers** [hashicorp/vault#5973]
**Describe the bug**
When using the PKI Engine with the Filesystem Storage Backend on Windows Server 2012R2 users are unable READ certificates after creation. However, users are still able to generate new certificates and are able to obtain the certificate at the time of generation.

Attempting to access a certificate after generation results in the following error message:

```
{
    "errors": [
        "error fetching certificate 46:ca:33:a2:53:d4:cc:86:ea:b5:3c:70:be:0b:f8:fb:83:2c:13:cd: open C:\\vault\\data\\logical\\06cc3646-6183-4192-e971-9c65b02c6049\\revoked\\_46:ca:33:a2:53:d4:cc:86:ea:b5:3c:70:be:0b:f8:fb:83:2c:13:cd: The filename, directory name, or volume label syntax is incorrect."
    ]
}
```

Upon inspecting the server it appears that the revoked directory is missing from the path. This is unusual for two reasons.

1. I assume VAULT is supposed to create these directories. This doesn't appear to be occuring.
2. The certificate has not been revoked - why is this folder necessary?

Server output:

```
    Directory: C:\Vault\Data\Logical\06cc3646-6183-4192-e971-9c65b02c6049


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       12/17/2018  11:18 AM                certs
d-----       12/17/2018  10:52 AM                config
d-----       12/17/2018  11:00 AM                role
-a----       12/17/2018  11:13 AM           1329 _ca
-a----       12/17/2018  11:13 AM            617 _crl
-a----       12/18/2018   9:17 AM            265 _urls
```

An additional difference I have noted:
The certificates stored in the `certs` directory above use dashes `-` instead of colons `:` in the filename. Perhaps this is an additional difference that's causing breakage?

**To Reproduce**
1. Install VAULT with the Filesystem Storage Backend on Windows Server 2012R2
2. Follow the guide here - https://learn.hashicorp.com/vault/secrets-management/sm-pki-engine and stop after `STEP 4 - Request Certificates.`
3. Attempt to READ the certificate that we created - https://www.vaultproject.io/api/secret/pki/index.html#read-certificate

**Expected behavior**
The certificate in PEM formatting in the certificate key of the JSON object is returned.

**Environment:**
* Vault Server Version (retrieve with `vault status`): Version         1.0.0
* Vault CLI Version (retrieve with `vault version`): Vault v1.0.0 ('c19cef14891751a23eaa9b41fd456d1f99e7e856')
* Server Operating System/Architecture: Windows Server 2012R2 Datacenter

Vault server configuration file(s):

```ui = true

storage "file" {
  path    = "C:/vault/data"
}

listener "tcp" {
 address = "0.0.0.0:8200"
 tls_disable = 0
 tls_cert_file = "C:/vault/cert/**********.cert.pem"
 tls_key_file = "C:/vault/cert/**********.key.pem"
 tls_client_ca_file = "C:/vault/cert/ca-chain.cert.pem"
 tls_min_version = "tls12"
 tls_disable_client_certs = 1
}

```

**Additional context**

**Comment by mfriesen08** [comment]
I can confirm I am also having the same issues retrieving certificates on vault version 1.0 with a file system backend.

**Comment by jefferai** [comment]
Probably Windows isn't liking that we look at paths with `:` for legacy reasons, which is how they used to be stored until we changed them to `-` for Windows reasons. What is interesting is we haven't changed this code in a long long time but after we moved to using `-` for PKI file paths for certs all the Windows complaints went away, and that was a good year and a half ago, so I would expect that Windows users would have complained in the interim, unless they simply turned off storage of certs altogether. Now suddenly two people are having this problem. Quite odd.

We should probably either allow an error in the legacy call, or skip the legacy call altogether on Windows.

**Comment by brrogers** [comment]
The above issue also prevents certificate revokation:

```
{
    "errors": [
        "1 error occurred:\n\t* error fetching certificate 46:ca:33:a2:53:d4:cc:86:ea:b5:3c:70:be:0b:f8:fb:83:2c:13:cd: open C:\\vault\\data\\logical\\06cc3646-6183-4192-e971-9c65b02c6049\\revoked\\_46:ca:33:a2:53:d4:cc:86:ea:b5:3c:70:be:0b:f8:fb:83:2c:13:cd: The filename, directory name, or volume label syntax is incorrect.\n\n"
    ]
}
```

Updated Vault server to 1.0.1 - No change in behavior.


Really crossing my fingers that this is fixed in 1.0.2

**Comment by jefferai** [comment]
I'm not sure if it will be by the Vault team but if you're motivated it is relatively straightforward.
